src/main/java/com/couchbase/lite/BlobStore.java

```diff
@@ -534,33 +469,30 @@ public boolean storeBlob(byte[] data, BlobKey outKey) {
             }
         }
 
-        FileOutputStream fos = null;
+        FileOutputStream fos;
         try {
             fos = new FileOutputStream(file);
-            fos.write(data);
         } catch (FileNotFoundException e) {
             Log.e(Log.TAG_DATABASE, "BlobStore: Error opening file for output", e);
             return false;
+        }
+        try {
+            fos.write(data);
         } catch (IOException ioe) {
             Log.e(Log.TAG_DATABASE, "BlobStore: Error writing to file", ioe);
             return false;
         } finally {
-            if (fos != null) {
-                try {
-                    fos.close();
-                } catch (IOException e) {
-                    // ignore
-                }
+            try {
+                fos.close();
+            } catch (IOException e) {
             }
         }
         return true;
     }
 
     private static byte[] getBytesFromFile(File file) throws IOException {
-        InputStream is = null;
+        InputStream is = new FileInputStream(file);
```

Comment:
The general best practice for native resource allocation is to minimize its lifespan: initialize just before you need it, and release it just after you finish with it. Why not move this line down to just before line 577/509, and reduce the scope of the `try`/`finally` block to just the `while` loop?